pilot/pkg/config/kube/ior/route.go

```diff
@@ -555,36 +290,41 @@ func (r *route) findService(gateway *networking.Gateway) (string, string, error)
 
 	for _, ns := range namespaces {
```

Comment:
I think you can get rid of this for loop and the `namespaces` list altogether. Instead, list the pods via the informer/lister (xns-informer will make sure you're listing pods only in the member namespaces). 

If you do this, IOR no longer needs to be a `MemberRollListener`.

Reply:
Removed usage of `MemberRoll`.